kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1288 @@
+                    KRATOS_ERROR <<"The equation system size has changed during the simulation. This is not permited."<<std::endl;
+                    rTMatrix.resize(BaseType::mEquationSystemSize, mDoFToSolveSystemSize, true);
+                    ConstructRelationMatrixStructure(pScheme, rTMatrix, rModelPart);
+                }
+            }
+        }
+    }
+
+    /**
+     * @brief This function is intended to be called at the end of the solution step to clean up memory storage not needed
+     */
+    void Clear() override
+    {
+        BaseType::Clear();
+
+        mDoFToSolveSet = DofsArrayType();
+        mSolvableDoFReorder.clear();
+        mSlaveMasterDoFRelation.clear();
+
+        // Set the flag
+        mCleared = true;
+
+        KRATOS_INFO_IF("ResidualBasedEliminationBuilderAndSolverWithConstraints", this->GetEchoLevel() > 1) << "Clear Function called" << std::endl;
+    }
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    TSystemMatrixPointerType mpTMatrix = NULL; /// This is matrix containing the global relation for the constraints
+
+    DofsArrayType mDoFToSolveSet;              /// The set containing the solvable DoF of the system
+
+    SizeType mDoFToSolveSystemSize = 0;        /// Number of degrees of freedom of the problem to actually be solved
+
+    bool mCleared = true; /// If the system has been reseted
+
+    std::unordered_map<IndexType, IndexType> mSolvableDoFReorder; /// The correlation between the global total DoF order and the solvable DoF order
+    std::unordered_map<IndexType, IndexSetType> mSlaveMasterDoFRelation; /// The relation between the slave and the master DoFs
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    /**
+     * @brief This method computes the equivalent coounter part of the SetUpSystem when using constraints
+     * @param rModelPart The model part of the problem to solve
+     */
+    void SetUpSystemWithConstraints(ModelPart& rModelPart)
+    {
+        KRATOS_TRY
+
+        // First we set up the system of equations without constraints
+        BaseType::SetUpSystem(rModelPart);
+
+        /// A pair of Ids
+        typedef std::pair<IndexType, IndexType> IdPairType;
+
+        // Add the computation of the global ids of the solvable dofs
+        mSolvableDoFReorder.reserve(mDoFToSolveSystemSize);
+        IndexType counter = 0;
+        for (auto& dof : BaseType::mDofSet) {
+            if (dof.EquationId() < BaseType::mEquationSystemSize) {
+                auto it = mDoFToSolveSet.find(dof);
+                if (it != mDoFToSolveSet.end()) {
+                    mSolvableDoFReorder.insert(IdPairType(dof.EquationId(), counter));
```

Comment:
as i am telling, keep the ordering here

Reply:
@RiccardoRossi I am unable to make this work with just a vector